### PR TITLE
Merge angles branch [DO NOT MERGE UNTIL TESTED]

### PR DESCRIPTION
**Pull request by whert-dev**

### CODE REVIEW

File: magic_numbers.py [hunk 1]
```diff
@@ -15,5 +16,31 @@
...
+
 # Factor by which areas of contours are multiplied to get an estimate for allowed distance squared between them
-METRIC_SCALE_FACTOR = 27.5
+METRIC_SCALE_FACTOR = 26
+
+# The following were careful measurements of the frame area with the camera
+# aimed at a flat wall, and the distance of the camera from the wall. All are in
+# millimetres.
+FOV_WIDTH = 1793
+FOV_HEIGHT = 2303
```
Comment (OliverW10) [inline magic_numbers.py:28]: we should double check these, it seems weird that there is a higher vertical fov that horizontal

File: vision.py [hunk 2]
```diff
@@ -126,31 +153,40 @@ def rank_groups(
...
+) -> float:
+    min_y = FRAME_HEIGHT
+    max_y = 0
+    for i in group:
+        c = contours[i]
+        for p in c:
+            max_y = max(max_y, p[0][1])
+            min_y = min(min_y, p[0][1])
+    height = max_y - min_y
+    return sum(contour_areas[i] for i in group) * TOTAL_AREA_K + height * HEIGHT_K
```
Comment (OliverW10) [inline vision.py:172]: should we be taking the sqrt of the total area?

File: magic_numbers.py [hunk 3]
```diff
@@ -31,13 +31,17 @@
 FOV_DISTANCE = 2234
 
 MAX_FOV_WIDTH = math.atan2(FOV_WIDTH / 2, FOV_DISTANCE) * 2  # 54.54 degrees
-MAX_FOV_HEIGHT = 45 / 180 * math.pi #math.atan2(FOV_HEIGHT / 2, FOV_DISTANCE) * 2  # 43.73 degrees
+MAX_FOV_HEIGHT = (
+    45 / 180 * math.pi
```
Comment (auscompgeek) [inline magic_numbers.py:35]: Use `math.radians` instead.

File: camera_manager.py [hunk 4]
```diff
@@ -43,6 +43,13 @@ def __init__(
         # parameters are the opposite of numpy's (technically it is an array, not an actual image).
         self.frame = np.zeros(shape=(FRAME_HEIGHT, FRAME_WIDTH, 3), dtype=np.uint8)
 
+        self.set_camera_property("white_balance_temperature_auto", 0)
+        self.set_camera_property("exposure_auto_priority", 0)
+        self.set_camera_property("exposure_auto", 1)
```
Comment (auscompgeek) [inline camera_manager.py:48]: surely this should be false?

File: test_images/expected_fail.csv [hunk 5]
```diff
@@ -1 +1,3 @@
-TarmacCenter2ft10in.png,-0.16562500000000002,-0.7388888888888889
+other/TarmacCenter2ft10in.png,-0.16562500000000002,-0.7388888888888889
+test_images/our_dataset/mpv-shot0001.jpg,0.0,0.0
+test_images/our_dataset/mpv-shot0010.jpg,0.006250000000000089,-0.19999999999999996
```
Comment (auscompgeek) [inline test_images/expected_fail.csv:3]:
```suggestion
our_dataset/mpv-shot0001.jpg,0.0,0.0
our_dataset/mpv-shot0010.jpg,0.006250000000000089,-0.19999999999999996
```

File: test_images/expected_fail.csv [hunk 6]
```diff
@@ -1 +1,3 @@
-TarmacCenter2ft10in.png,-0.16562500000000002,-0.7388888888888889
+other/TarmacCenter2ft10in.png,-0.16562500000000002,-0.7388888888888889
```
Comment (auscompgeek) [inline test_images/expected_fail.csv:1]: this seems to refer to a file that no longer exists?

File: magic_numbers.py [hunk 7]
```diff
@@ -15,5 +16,45 @@
...
+TRIG_DISTANCE_K = 0.5
+AREA_DISTANCE_K = 1 - TRIG_DISTANCE_K
+
+# Coeffecients for group fitness
+HEIGHT_K = -0.1
+TOTAL_AREA_K = 1
+
+# Coeffecients for group confidence
+CONF_ASPECT_RATIO_K = 2
+CONF_RECTANGULARES_K = 3
```
Comment (auscompgeek) [inline magic_numbers.py:57]: typo? I'm not sure what this word should be though

**Review by auscompgeek — APPROVED** [review]
lgtm otherwise

**Pull request merged**